template/template.go

```diff
@@ -76,3 +84,137 @@ func (t *Template) ExecuteHTMLString(html string, data interface{}) (string, err
 	err = tmpl.Execute(&buf, data)
 	return buf.String(), err
 }
+
+type FuncMap map[string]interface{}
+
+var DefaultFuncs = FuncMap{
+	"upper": strings.ToUpper,
+	"lower": strings.ToLower,
+	"title": strings.Title,
+	// pairs allows for in-order iteration of key/value pairs.
+	"pairs": func(m map[string]string) []Pair {
+		var (
+			pairs     = make([]Pair, 0, len(m))
+			keys      = make([]string, 0, len(m))
+			sortStart = 0
+		)
+		for k := range m {
+			if k == string(model.AlertNameLabel) {
+				keys = append([]string{k}, keys...)
+				sortStart = 1
+			} else {
+				keys = append(keys, k)
+			}
+		}
+		sort.Strings(keys[sortStart:])
+
+		for _, k := range keys {
+			pairs = append(pairs, Pair{k, m[k]})
+		}
+		return pairs
+	},
+	"firing": func(alerts []Alert) []Alert {
```

Comment:
Thinking a bit and considering how the system is designed, I think the sanest thing is to add an options to all the notification methods where this is not something ~everyone would want (i.e. not Pagerduty) to filter out the resolved alerts. The default would be to filter out the resolved alerts.

We'd still need the default templates to handle all this in case someone flipped the option, but it'll put most users on the right path.